src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -437,12 +439,14 @@ exprt string_constraint_generatort::add_axioms_for_parse_int(
     not_exprt(or_exprt(starts_with_minus, starts_with_plus));
 
   /// TODO: we should throw an exception when this does not hold:
-  exprt correct=add_axioms_for_correct_number_format(str, radix);
+  std::size_t max_string_length=40;
```

Comment:
Sure. 40 is just a number I chose so that we can parse some longs in binary that wouldn't fit into an int but it isn't too slow. To be able to parse all valid longs in binary, assuming no leading zeroes, we need at least 65 characters. Hopefully diffblue/test-gen#702 will help a lot with this.